src/aolayer.cpp

```diff
@@ -285,6 +285,7 @@ void AOLayer::start_playback(QString p_image)
     frame = 0;
     continuous = false;
   }
+  // FIXME: this causes a hitch, find a more performant alternative
```

Comment:
You've caught my attention with this comment...

You are doing too many copies, and `movie_frames` needs to be resized ahead of time to accommodate the number of frames you are storing - otherwise it will resize over and over.

Reply:
Presently unfixable.
The correct way of doing this would be to use QImageReader::jumpToImage() and populate missing data in the movie ticker when it's needed. This is unforunately completely impossible, because QImageReader::jumpToImage() is not implemented in any image format AO2 is equipped to use. Instead, the default behavior is used - that is, absolutely nothing.

I need a drink.